homeassistant/components/wevolor/cover.py

```diff
@@ -0,0 +1,80 @@
+"""Cover component for shades controlled by the Wevolor controller."""
+
+from __future__ import annotations
+
+from pywevolor import Wevolor
+
+from homeassistant.components.cover import (
+    SUPPORT_CLOSE,
+    SUPPORT_CLOSE_TILT,
+    SUPPORT_OPEN,
+    SUPPORT_OPEN_TILT,
+    SUPPORT_STOP,
+    SUPPORT_STOP_TILT,
+    CoverDeviceClass,
+    CoverEntity,
+)
+
+from .const import CONFIG_CHANNELS, CONFIG_TILT, DOMAIN
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Wevolor shades."""
+
+    wevolor = hass.data[DOMAIN][config_entry.entry_id]
+
+    entities = [
+        WevolorShade(wevolor, i, config_entry.data[CONFIG_TILT])
+        for i in range(1, config_entry.data[CONFIG_CHANNELS] + 1)
+    ]
+    async_add_entities(entities, True)
```

Comment:
Should not update_before_add.
```suggestion
    async_add_entities(entities)
```